report_templates/host_-_last_checkin.erb

```diff
@@ -0,0 +1,28 @@
+<%#
+name: Host - Last Checkin
+snippet: false
+template_inputs:
+- name: Inactive for
+  required: false
+  input_type: user
+  description: A time internval after which the Host is considered inactive. It can
+    be any search syntax supported interval, e.g. "4 hours", "30 minutes", "2 days"
+  advanced: false
+  value_type: plain
+  resource_type: Katello::ActivationKey
+  default: 5 hours
+  hidden_value: false
+model: ReportTemplate
+require:
+- plugin: katello
+￼ version: 3.16.0
+%>
+<%- load_hosts(search: "last_checkin < \"#{input('Inactive for')} ago\"").each_record do |host| -%>
+<%- report_headers('Name', 'IP', 'Last Checkin') -%>
+<%-   report_row(
+        'Name': host.name,
+        'IP': host.ip,
+        'Last Checkin': last_checkin(host),
```

Comment:
Does this result in a 1+n DB queries or is it more of a formatting function?

Reply:
good catch, I meant to check n+1 but forgot, I added `preload` now and also loaded NICs, which also suffered from the same problem